qa/SidechainTestFramework/scutil.py

```diff
@@ -81,153 +88,284 @@ def sync_sc_mempools(api_connections, wait_for=25):
+        "mcBlockHeight": 221,
+        "mcNetwork": "regtest"
+    }
+
+    genesis_secrets = None
+
+    if(account_secrets==None):
+        genesis_secrets = {0 : "6882a61d8a23a9582c7c7e659466524880953fa25d983f29a8e3aa745ee6de5c0c97174767fd137f1cf2e37f2e48198a11a3de60c4a060211040d7159b769266", \
+                           1 : "905e2e581615ba0eff2bcd9fb666b4f6f6ed99ddd05208ae7918a25dc6ea6179c958724e7f4c44fd196d27f3384d2992a9c42485888862a20dcec670f3c08a4e", \
+                           2 : "80b9a06608fa5dbd11fb72d28b9df49f6ac69f0e951ca1d9e67abd404559606be9b36fb5ae7e74cc50603b161a5c31d26035f6a59e602294d9900740d6c4007f"}
+    else:
+        genesis_secrets = []
+        for i in range(len(account_secrets)):
+            genesis_secrets.append(str(account_secrets[i]["secret"]))
+
     apiAddress = "127.0.0.1"
-    with open('./resources/template.conf','r') as templateFile: 
+    with open('./resources/template.conf', 'r') as templateFile:
         tmpConfig = templateFile.read()
     configsData = []
     apiPort = sc_rpc_port(n)
     bindPort = sc_p2p_port(n)
-    datadir = os.path.join(dirname, "sc_node"+str(n))
+    datadir = os.path.join(dirname, "sc_node" + str(n))
     if not os.path.isdir(datadir):
         os.makedirs(datadir)
     config = tmpConfig % {
-        'NODE_NUMBER' : n,
-        'DIRECTORY' : dirname,
-        'WALLET_SEED' : ("minerNode" + str(n+1)) if n < 3 else ("node" + str(n+1)),
-        'API_ADDRESS' : "127.0.0.1",
-        'API_PORT' : str(apiPort),
-        'BIND_PORT' : str(bindPort),
-        'OFFLINE_GENERATION' : "false",
-        'GENESIS_DATA': "" if genesisData is None else str(genesisData),
-        'GENESIS_SECRETS': genesis_secrets[n]
-        }
+        'NODE_NUMBER': n,
+        'DIRECTORY': dirname,
+        'WALLET_SEED': "sidechain_seed_{0}".format(n),
+        'API_ADDRESS': "127.0.0.1",
+        'API_PORT': str(apiPort),
+        'BIND_PORT': str(bindPort),
+        'OFFLINE_GENERATION': "false",
+        'GENESIS_SECRETS': genesis_secrets[n] if account_secrets==None else ','.join(['"{0}"'.format(value) for value in genesis_secrets])[1:-1],
+        'SIDECHAIN_ID': jsonNode["scId"],
+        'GENESIS_DATA': jsonNode["scGenesisBlockHex"],
+        'POW_DATA': jsonNode["powData"],
+        'BLOCK_HEIGHT': jsonNode["mcBlockHeight"],
+        'NETWORK': str(jsonNode["mcNetwork"])
+    }
     configsData.append({
-        "name" : "node" + str(n),
-        "url" : "http://" + apiAddress + ":" + str(apiPort)
+        "name": "node" + str(n),
+        "url": "http://" + apiAddress + ":" + str(apiPort)
     })
-    with open(os.path.join(datadir, "node"+str(n)+".conf"), 'w+') as configFile:
+    with open(os.path.join(datadir, "node" + str(n) + ".conf"), 'w+') as configFile:
         configFile.write(config)
     return configsData
 
+
+def sc_generate_genesis_data(self):
```

Comment:
do we need this here? moreover do we need this at all?

Reply:
No we don't need it.
I left it for don't create side effects before the code review.
I'll remove it

Reply:
Removed